jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/internal/HttpChannelState.java

```diff
@@ -955,6 +956,12 @@ public void addHttpStreamWrapper(Function<HttpStream, HttpStream> wrapper)
             getHttpChannel().addHttpStreamWrapper(wrapper);
         }
 
+        @Override
+        public Session getSession(boolean create)
+        {
+            return null;
```

Comment:
Why is this not `getRequest().getSession(create)`?

Reply:
This is the base Request impl, that does not support sessions unless it is wrapped to do so.  There is no request below/under/inside this request to get.